packages/astro/components/index.js

```diff
@@ -1,2 +1,17 @@
-export { default as Markdown } from './Markdown.astro';
-export { default as Prism } from './Prism.astro';
+// `lazy` is a wrapper that creates async Astro components
+const lazy = (_import) => {
+  return {
+    isAstroComponent: true,
+    __render: async (...args) =>
+      _import().then(({ default: Component }) => {
+        return Component.__render(...args);
+      }),
+  };
+};
+
+// We use `lazy` to avoid the Astro Snowpack plugin from eagerly evaluating these components
+// because `__renderPage` is stateful and will inject any linked styles to `state`
+// for any consumers using `import { SingleComponent } from 'astro/component'`
+export const Markdown = lazy(() => import('./Markdown.astro'));
+export const Prism = lazy(() => import('./Prism.astro'));
+export const Debug = lazy(() => import('./Debug.astro'));
```

Comment:
Here is that bugfix! Turns out using
```
export { default as Debug } from './Debug.astro';
```
was being evaluated eagerly by the Astro Snowpack plugin, which meant that the codegen `state` was picking up CSS from _any_ component exported here.

The fix here is to make each component lazy, so they are only compiled when they are actually rendered.

Reply:
:0 so that was a problem with re-exporting them into the same file? Could this happen with other projects?

Reply:
Opened #701 to track this. TBD on how to proceed here.